--- buildtools/wafsamba/wafsamba.py	2015-02-24 19:59:51.000000000 +0100
+++ buildtools/wafsamba/wafsamba.py	2015-03-23 01:01:20.239995548 +0100
@@ -763,7 +763,7 @@
         installed_file.write(newline)
         lineno = lineno + 1
     installed_file.close()
-    os.chmod(installed_location, 0755)
+    os.chmod(installed_location, 0o755)
     return 0
 
 def copy_and_fix_perl_path(task):
@@ -791,7 +791,7 @@
         installed_file.write(newline)
         lineno = lineno + 1
     installed_file.close()
-    os.chmod(installed_location, 0755)
+    os.chmod(installed_location, 0o755)
     return 0
 
 
