--- buildtools/wafsamba/samba_dist.py   2015-03-23 01:49:39.399999849 +0100
+++ buildtools/wafsamba/samba_dist.py   2014-10-01 11:16:21.000000000 +0200
@@ -171,7 +171,7 @@ def dist(appname='', version=''):
         absdir = os.path.join(srcdir, dir)
         try:
             files = vcs_dir_contents(absdir)
-        except Exception as e:
+        except Exception, e:
             Logs.error('unable to get contents of %s: %s' % (absdir, e))
             sys.exit(1)
         add_files_to_tarball(tar, srcdir, dir, dist_base, destdir, blacklist, files)
