--- buildtools/wafsamba/samba_conftests.py	2015-02-24 19:59:51.000000000 +0100
+++ buildtools/wafsamba/samba_conftests.py	2015-03-23 01:17:52.369995169 +0100
@@ -51,7 +51,7 @@
     ret = None
     try:
         ret = self.run_c_code(*k, **kw)
-    except Configure.ConfigurationError, e:
+    except Configure.ConfigurationError as e:
         self.check_message_2(kw['errmsg'], 'YELLOW')
         if 'mandatory' in kw and kw['mandatory']:
             if Logs.verbose > 1:
