--- buildtools/wafadmin/Configure.py	2014-10-01 11:16:21.000000000 +0200
+++ buildtools/wafadmin/Configure.py	2015-03-23 00:31:28.239996231 +0100
@@ -176,7 +176,7 @@
 			module = None
 			try:
 				module = Utils.load_tool(tool, tooldir)
-			except Exception, e:
+			except Exception as e:
 				ex = e
 				if Options.options.download:
 					_3rdparty = os.path.normpath(Options.tooldir[0] + os.sep + '..' + os.sep + '3rdparty')
