--- wscript	2014-12-20 16:07:55.000000000 +0100
+++ wscript	2015-03-23 00:55:29.559995681 +0100
@@ -202,7 +202,7 @@
                 need_pie = False
         if conf.check_cc(cflags='-fPIE', ldflags='-pie', mandatory=need_pie,
                          msg="Checking compiler for PIE support"):
-		conf.env['ENABLE_PIE'] = True
+                         conf.env['ENABLE_PIE'] = True
 
     if Options.options.enable_relro != False:
         if Options.options.enable_relro == True:
