--- buildtools/wafadmin/Tools/compiler_cc.py	2015-03-23 01:49:10.989999860 +0100
+++ buildtools/wafadmin/Tools/compiler_cc.py	2014-10-01 11:16:21.000000000 +0200
@@ -39,7 +39,7 @@ def detect(conf):
 		conf.env = orig.copy()
 		try:
 			conf.check_tool(compiler)
-		except Configure.ConfigurationError as e:
+		except Configure.ConfigurationError, e:
 			debug('compiler_cc: %r' % e)
 		else:
 			if conf.env['CC']:
--- buildtools/wafsamba/samba_dist.py	2015-03-23 01:49:39.399999849 +0100
+++ buildtools/wafsamba/samba_dist.py	2014-10-01 11:16:21.000000000 +0200
@@ -171,7 +171,7 @@ def dist(appname='', version=''):
         absdir = os.path.join(srcdir, dir)
         try:
             files = vcs_dir_contents(absdir)
-        except Exception as e:
+        except Exception, e:
             Logs.error('unable to get contents of %s: %s' % (absdir, e))
             sys.exit(1)
         add_files_to_tarball(tar, srcdir, dir, dist_base, destdir, blacklist, files)
--- buildtools/wafsamba/wafsamba.py	2015-03-23 02:08:04.499999427 +0100
+++ buildtools/wafsamba/wafsamba.py	2015-03-23 01:45:45.429999938 +0100
@@ -953,6 +953,10 @@ def samba_display(self):
         return progress_display(self, ext_map[ext], fname)
     return Task.Task.old_display(self)
 
+Task.TaskBase.classes['Task'].old_display = Task.TaskBase.classes['Task'].display
+Task.TaskBase.classes['Task'].display = samba_display
+
+
 @after('apply_link')
 @feature('cshlib')
 def apply_bundle_remove_dynamiclib_patch(self):
--- buildtools/wafsamba/wscript	2015-03-23 01:59:49.679999616 +0100
+++ buildtools/wafsamba/wscript	2015-02-24 19:59:51.000000000 +0100
@@ -495,7 +495,7 @@ struct foo bar = { .y = 'X', .x = 1 };
     # Extra sanity check.
     if conf.CONFIG_SET("HAVE_BIG_ENDIAN") == conf.CONFIG_SET("HAVE_LITTLE_ENDIAN"):
         Logs.error("Failed endian determination.  The PDP-11 is back?")
-        sys.exit(1)
+	sys.exit(1)
     else:
         if conf.CONFIG_SET("HAVE_BIG_ENDIAN"):
             conf.DEFINE('WORDS_BIGENDIAN', 1)
--- configure	2015-03-23 02:07:27.379999442 +0100
+++ configure	2014-10-01 11:16:21.000000000 +0200
@@ -10,5 +10,5 @@ JOBS=1
 export JOBS
 
 cd . || exit 1
-${PYTHON:=python2} $WAF configure "$@" || exit 1
+${PYTHON:=python} $WAF configure "$@" || exit 1
 cd $PREVPATH
