--- buildtools/wafadmin/Task.py	2014-10-01 11:16:21.000000000 +0200
+++ buildtools/wafadmin/Task.py	2015-03-23 01:14:35.929995244 +0100
@@ -145,7 +145,7 @@
 	def set_group(self, idx):
 		if isinstance(idx, str):
 			g = self.groups_names[idx]
-			for x in xrange(len(self.groups)):
+			for x in range(len(self.groups)):
 				if id(g) == id(self.groups[x]):
 					self.current_group = x
 		else:
@@ -271,9 +271,9 @@
 		keys = self.cstr_groups.keys()
 		max = len(keys)
 		# hopefully the length of this list is short
-		for i in xrange(max):
+		for i in range(max):
 			t1 = self.cstr_groups[keys[i]][0]
-			for j in xrange(i + 1, max):
+			for j in range(i + 1, max):
 				t2 = self.cstr_groups[keys[j]][0]
 
 				# add the constraints based on the comparisons
@@ -795,7 +795,7 @@
 		debug("Task %r", self)
 		msgs = ['Task must run', '* Source file or manual dependency', '* Implicit dependency', '* Environment variable']
 		tmp = 'task: -> %s: %s %s'
-		for x in xrange(len(msgs)):
+		for x in range(len(msgs)):
 			if (new_sigs[x] != old_sigs[x]):
 				debug(tmp, msgs[x], v(old_sigs[x]), v(new_sigs[x]))
 
@@ -999,7 +999,7 @@
 	buf = []
 	dvars = []
 	app = buf.append
-	for x in xrange(len(extr)):
+	for x in range(len(extr)):
 		params[x] = params[x].strip()
 		if params[x]:
 			app("lst.extend(%r)" % params[x].split())
